vowpalwabbit/arguments.h

```diff
@@ -0,0 +1,157 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <typeinfo>
+#include <memory>
+
+namespace VW {
+
+  struct base_argument {
+    base_argument(std::string name, size_t type_hash)
+      : m_name(name), m_type_hash(type_hash)
+    {}
+
+    std::string m_name;
+    size_t m_type_hash;
+    std::string m_help = "";
+    std::string m_short_name = "";
+    bool m_keep = false;
+
+    virtual ~base_argument() {}
+  };
+
+  template<typename T>
+  struct typed_argument : base_argument {
+    typed_argument(std::string name, T* location)

```

Comment:
Maybe use T& here?  There is no need to do pointer operations.

Reply:
Good point